### PR TITLE
backend/local: batch split region with batch limit

### LINKED ISSUE #481: Lightning needs to set an upper limit of split keys in one BatchSplit RPC call
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

Lightning needs to set an upper limit of split keys in one BatchSplit RPC call. TiKV has a hard size limit of a raft log, too many split keys in one request could fail batch split.

2. What did you expect to see?

There is an upper limit of split keys in one BatchSplit RPC call.

3. What did you see instead?

No limit.

4. Versions of the cluster

master

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Limit the batch split key size when batch size regions

close #481 

Need https://github.com/pingcap/br/pull/612 to be merged first.

### What is changed and how it works?
- Run batch split in a loop to limit each request's key size
- Add retry for pagnateScanRegion and scatter region 
- Add some log 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

### CODE REVIEW

**Review by overvenus — COMMENTED** [review]
Could you add a unit test?

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: lightning/backend/localhelper.go [hunk 1]
```diff
@@ -61,26 +83,49 @@ func (local *local) SplitAndScatterRegionByRanges(ctx context.Context, ranges []
...
-					for _, key := range keys {
-						log.L().Warn("no valid key",
-							zap.Binary("startKey", region.Region.StartKey),
-							zap.Binary("endKey", region.Region.EndKey),
-							zap.Binary("key", codec.EncodeBytes([]byte{}, key)))
+			sort.Slice(keys, func(i, j int) bool {
+				return bytes.Compare(keys[i], keys[j]) < 0
+			})
+			splitRegion := region
+			for i := 0; i < (len(keys)+maxBatchSplitKeys)/maxBatchSplitKeys; i++ {
```
Comment (3pointer) [inline lightning/backend/localhelper.go:93]: rename`i` is better.

Comment (3pointer) [reply lightning/backend/localhelper.go:93]: if  len(keys)= maxBatchSplitKeys it seems have some problems.

File: lightning/backend/localhelper_test.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
+package backend
```
Comment (3pointer) [inline lightning/backend/localhelper_test.go:1]: need License

Comment (glorv) [reply lightning/backend/localhelper_test.go:14]: done

**Review by 3pointer — COMMENTED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]

**Pull request merged**